dsls/sql/src/main/java/org/apache/beam/dsls/sql/schema/BeamSQLRow.java

```diff
@@ -145,13 +154,19 @@ public Object getFieldValue(int fieldIdx) {
 
     switch (fieldType) {
     case INTEGER:
+      if (!(fieldValue instanceof Integer)) {
+        throw new InvalidFieldException(
+            String.format("[%s] doesn't match type [%s]", fieldValue, fieldType));
+      } else {
+        return Integer.valueOf(fieldValue.toString());
```

Comment:
It seems that `XXX.valueOf()` calls in `getFieldValue` is not necessary? Because the return value of `getFieldValue` is `Object`, and since the value and type are checked in `addField()` method, so type cast in methods like `getShort`, `getInteger` are safe?